packages/sdk-react/src/core/queries/useCounterpart.ts

```diff
@@ -128,251 +95,145 @@ export const counterpartQueryKeys = {
-  useEntityListCache<CounterpartResponse>(counterpartQueryKeys.list);
-
-const useCounterpartDetailCache = () =>
-  useEntityCache<CounterpartResponse>(counterpartQueryKeys.detail);
-
-const useCounterpartAddressListCache = (id: string) =>
-  useEntityListCache<CounterpartAddressResponseWithCounterpartID>(() => [
-    ...counterpartQueryKeys.addressList(id),
-  ]);
-
-const useCounterpartBankListCache = (id: string) =>
-  useEntityListCache<CounterpartBankAccountResponse>(() => [
-    ...counterpartQueryKeys.bankList(id),
-  ]);
-
-const useCounterpartContactListCache = (counterpartId: string) =>
-  useEntityListCache<CounterpartContactResponse>(() => [
-    ...counterpartQueryKeys.contactList(counterpartId),
-  ]);
-
-const useCounterpartVatListCache = (counterpartId: string) =>
-  useEntityListCache<CounterpartVatIDResponse>(() => [
-    ...counterpartQueryKeys.vatList(counterpartId),
-  ]);
-
 export const useCounterpartAddresses = (counterpartId?: string) => {
-  const { monite } = useMoniteContext();
-
-  return useQuery<CounterpartAddressResponseWithCounterpartID[], Error>({
-    queryKey: counterpartQueryKeys.addressList(counterpartId),
-
-    queryFn: () =>
-      !!counterpartId
-        ? monite.api.counterpartsAddresses
-            .getCounterpartAddresses(counterpartId)
-            .then((response) => response.data)
-        : [],
-
-    enabled: !!counterpartId,
-  });
-};
+  const { api } = useMoniteContext();
 
-export const useCounterpartAddressById = (
-  counterpartId?: string,
-  addressId?: string
-) => {
-  const { monite } = useMoniteContext();
-
-  return useQuery<
-    CounterpartAddressResponseWithCounterpartID | undefined,
-    Error
-  >({
-    queryKey: counterpartQueryKeys.addressDetail(counterpartId!, addressId!),
-
-    queryFn: () => {
-      if (!counterpartId || !addressId) {
-        throw new Error(
-          'MoniteReactSDK: counterpartId and addressId required.'
-        );
-      }
-
-      return monite.api.counterpartsAddresses.getCounterpartAddressesById(
-        addressId,
-        counterpartId
-      );
+  return api.counterparts.getCounterpartsIdAddresses.useQuery(
+    {
+      path: { counterpart_id: counterpartId ?? '' },
     },
-
-    enabled: Boolean(counterpartId) && Boolean(addressId),
-  });
+    {
+      enabled: !!counterpartId,
+    }
+  );
 };
 
-export const useUpdateCounterpartAddress = (
-  addressId: string,
-  counterpartId: string
-) => {
+export const useUpdateCounterpartAddress = () => {
   const { i18n } = useLingui();
-  const { monite } = useMoniteContext();
-  const { update } = useCounterpartAddressListCache(counterpartId);
-
-  return useMutation<
-    CounterpartAddressResponseWithCounterpartID,
-    ApiError,
-    CounterpartUpdateAddress
-  >({
-    mutationFn: (payload) =>
-      monite.api.counterpartsAddresses.updateCounterpartsAddress(
-        addressId,
-        counterpartId,
-        payload
-      ),
-
-    onSuccess: (address) => {
-      update(address);
```

Comment:
`update(address);` is `setQueryData(...)` and important, you can't just omit its usage.